src/Geta.EPi.Extensions/ContentReferenceExtensions.cs

```diff
@@ -135,7 +135,15 @@ public static string GetFriendlyUrl(this ContentReference contentReference, stri
                 ? urlResolver.GetUrl(contentReference, language, new VirtualPathArguments { ContextMode = ContextMode.Default })
                 : urlResolver.GetUrl(contentReference, language);
 
-            return includeHost ? url.AddHost() : url.RemoveHost();
+            if (includeHost)
+            {
+                return url.AddHost();
+            }
+
+            var content = Get<IContent>(contentReference);
+            var pageData = content as PageData;
```

Comment:
Why page data? What if it is a media content or other content?

Reply:
then its null, and there is no external url

Reply:
So only page data can have external links?

Reply:
yes